packages/cashscript/src/Transaction.ts

```diff
@@ -62,45 +63,60 @@ export class Transaction {
     private selector?: number,
   ) {}
 
-  from(input: Utxo): this;
-  from(inputs: Utxo[]): this;
+  from(input: ProvidedUtxo): this;
+  from(inputs: ProvidedUtxo[]): this;
 
-  from(inputOrInputs: Utxo | Utxo[]): this {
+  from(inputOrInputs: ProvidedUtxo | ProvidedUtxo[]): this {
     if (!Array.isArray(inputOrInputs)) {
       inputOrInputs = [inputOrInputs];
     }
 
-    this.inputs = this.inputs.concat(inputOrInputs);
+    const normalisedUtxos = inputOrInputs.map(
+      (utxo) => ({ ...utxo, satoshis: Number(utxo.satoshis) }),
+    );
```

Comment:
If I understand correctly, this conversion should never be an issue because even 21m e8 sats is within integer precision range for js floats. Is that right?

Reply:
Yes exactly

```js
MAX_SAFE_INTEGER = 9_007_199_254_740_991
MAX_SATS         = 2_100_000_000_000_000
```